src/Sets/Star.jl

```diff
@@ -166,3 +167,44 @@ Return the predicate of a star.
 A polyhedral set representing the predicate of the star.
 """
 predicate(X::STAR) = set(X)
+
+function _intersection_star!(c, V, P::Union{HPoly, HPolygon, HPolygonOpt}, H::HalfSpace)
+    a′ = transpose(V) * H.a
+    b′ = H.b - dot(H.a, c)
+    H′ = HalfSpace(a′, b′)
+    return addconstraint!(P, H′)
+end
+
+function intersection!(X::STAR, H::HalfSpace)
+    _intersection_star!(center(X), basis(X), predicate(X), H)
+    return X
+end
+
+function intersection(X::STAR{N, VN, MN, PT}, H::HalfSpace) where {N, VN, MN, PT<:Union{HPoly, HPolygon, HPolygonOpt}}
+    return intersection!(copy(X), H)
```

Comment:
```suggestion
    Y = copy(X)
    return intersection!(Y, H)
```
The problem is that `copy` returns the old object.

Reply:
i think we should add (if there doesn't exist one) a conversion to `HPolyhedron` with `Vector` normals

Reply:
Okay, but this refers to the other comment, right? :wink:

Reply:
yes :)